hs/server.hs

```diff
@@ -69,17 +79,38 @@ main = do
              dir graph $ graphResponse,
              dir code $ seeOther fbAuth1Url $ toResponse test,
              dir test $ look "code" >>= getEmail,
+             dir testPost $ look "code" >>= postToFacebook,
              dir about $ aboutResponse,
              dir static $ serveDirectory EnableBrowsing [] staticDir,
-             dir course $ path (\s -> liftIO $ queryCourse s)
+             dir course $ path (\s -> liftIO $ queryCourse s),
+             dir postFB $ seeOther fbAuth1UrlPost $ toResponse test
            ]
 
+postToFacebook :: String -> ServerPart Response
+postToFacebook code = (liftIO $ performPost code) >> graphResponse
+
+args2 :: FB.Argument
+args2 = ("message", "Test post please ignore")
+
+performPost :: String -> IO Response
+performPost code = withManager $ \manager -> FB.runFacebookT app manager $ do
```

Comment:
@david-yz-liu I had initially thought that you were referring to the line that you addressed, however I refactored based on the content of your comment. Is this what you had in mind, or would you rather refactor `withManager $ \manager -> FB.runFacebookT app manager $ do`? The latter probably makes a lot of sense, however it will require some deeper thought.